File: 125.验证回文串.cpp
/*
 * @lc app=leetcode.cn id=125 lang=cpp
 *
 * [125] 验证回文串
 */

// @lc code=start
class Solution {
public:
   bool isPalindrome(string s) {
        int len=s.size();
        string strs;
    
        for(int i=0;i<len;i++)
        {
            if(s[i]<='9'&&s[i]>='0'||s[i]>='a'&&s[i]<='z'||s[i]<='Z'&&s[i]>='A')
            {
                if(s[i]>='A'&&s[i]<='Z'){
                    s[i]+=32;
                }
                strs+=s[i];
            }
            else{
                continue;
            }
        }
        bool flag=true;
        for(int i=0;i<strs.size()/2;i++)
        {
            if(strs[i]!=strs[strs.size()-i-1])
            {
                flag=false;
            }

        }
        return flag;

    }
};
// @lc code=end

File: 125.验证回文串.py
#
# @lc app=leetcode.cn id=125 lang=python3
#
# [125] 验证回文串
#

# @lc code=start
def filter_string(s:str)->str:
        filtered_string=""
        for str in s:
            if str <='z' and str>='a' or str<='9'and str>='0' or str>='A' and str<='Z':
                filtered_string+=str
        return filtered_string
class Solution:
    
        
    def isPalindrome(self, s: str) -> bool:
        filtered_string=filter_string(s)
        reversed_string=filtered_string[::-1]
        return filtered_string.lower()==reversed_string.lower()

        


# @lc code=end

